### DISCUSSION TITLE
Add gitignore which included all auto-generated files

### DISCUSSION THREAD

**Discussion by iDoka** [esynr3z/corsair#100] [Questions] [answered]
Based on good practice of usage auto-generated source files in VCS should store only sources ("input" to generate some auto-generated entities), therefore  auto-generated sources should not added under VCS control. To avoid this, we must also generate a `.gitignore` file which included all auto-generated files.

Most convenient way to do this required to add new variable to `csrconfig` in  `[globcfg]` section like `BASEDIR` - the root path for other sections. All another `path` variables from secondary sections should complement  `BASEDIR` path to get full path.

In this case we can place `.gitignore` in  `BASEDIR` and place path and files into `.gitignore` relative to `BASEDIR`.

I think this is the easiest way to achieve goal:  auto-generated files cannot accidentally add under VCS control. 

 would be great to automaticaly add informative header to all auto-generated entities as comment:

**Comment by esynr3z** [comment]
>Based on good practice of usage auto-generated source files in VCS should store only sources ("input" to generate some auto-generated entities), therefore auto-generated sources should not added under VCS control.

Agree.

> To avoid this, we must also generate a .gitignore file which included all auto-generated files.

Debatable. Another good practice is to avoid any in-source file generation, so generated files are not mixed with the sources. That way you do not need precise gitignore to filter out artifacts. 
Usually it means that there is some output directory where all output goes to. And when you have such a directory it is super easy to append a single line to your own gitignore with it's name or pattern. 

Anyway, for Corsair 2.0 I want to force out-of-source generation making default behaviour such that all output goes to some "corsair-out" directory. As for your proposal, it would be also easy to put `.gitignore` with `*` glob inside to make sure this will never get into VCS.

**Comment by esynr3z** [comment, marked as answer]
In corsair v2.x gitignore for all files is automatically added to a build directory
